sympy/matrices/matrices.py

```diff
@@ -189,6 +189,9 @@ def __neg__(self):
         return -1*self
 
     def _matrix_pow_by_jordan_blocks(self, num):
+        if (num < 0) == True and self.det() == 0:
```

Comment:
I'm not sure that `jordan_cells()` has a standardized output for blocks corresponding to zero eigenvalues:

```python

In [15]: m = Matrix([[1, 1, 1], [3, 3, 3], [2, 2, 2]])

In [16]: m
Out[16]: 
⎡1  1  1⎤
⎢       ⎥
⎢3  3  3⎥
⎢       ⎥
⎣2  2  2⎦

In [17]: m.jordan_cells()
Out[17]: 
⎛⎡-1  -1  1/2⎤, [[0], [0], [6]]⎞
⎜⎢           ⎥                 ⎟
⎜⎢1   0   3/2⎥                 ⎟
⎜⎢           ⎥                 ⎟
⎝⎣0   1    1 ⎦                 ⎠
```

In this case it's `[0], [0]`, but it could be
```
[[0, 0],
 [0, 0]]
```